Whitelist-Dapp/hardhat-tutorial/contracts/Whitelist.sol

```diff
@@ -0,0 +1,39 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+
+contract Whitelist {
+
+    // Max number of whitelisted addresses allowed
+    uint8 public maxWhitelistedAddresses;
+
+    // Create a mapping of whitelistedAddresses
+    // if an address is whitelisted, we would set it to true, it is false by default for all other addresses.
+    mapping(address => bool) public whitelistedAddresses;
+
+    // numAddressesWhitelisted would be used to keep track of how many addresses have been whitelisted
+    // NOTE: Don't change this variable name, as it will be part of verification
+    uint8 public numAddressesWhitelisted;
+
+    // Setting the Max number of whitelisted addresses
+    // User will put the value at the time of deployment
+    constructor(uint8 _maxWhitelistedAddresses) {
```

Comment:
Better to initialise maxWhitelistedAddresses in field decleration to avoid mistake from the SC deployer. + This variable should be declared as "constant"